kpi-tracker/src/main.rs

```diff
@@ -407,24 +416,77 @@ async fn process_genesis_block(
     Ok(())
 }
 
-/// Process a block, represented by `block_hash`, updating the `db`
-/// corresponding to events captured by the block.
-async fn process_block(
+/// If block specified by `block_hash` is a payday block (also implies >=
+/// protocol version 4), this returns the total stake for that block. Otherwise
+/// returns `None`.
+async fn p4_payday_total_stake(
     node: &mut Client,
     block_hash: BlockHash,
-    db: &mut DB,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<Option<Amount>> {
+    let tokenomics_info = node
+        .get_tokenomics_info(block_hash)
+        .await
+        .with_context(|| format!("Could not get tokenomics info for block: {}", block_hash))?
+        .response;
+
+    if let RewardsOverview::V1 {
+        total_staked_capital,
+        ..
+    } = tokenomics_info
+    {
+        let mut special_events = node
+            .get_block_special_events(block_hash)
+            .await
+            .with_context(|| format!("Could not get special events for block: {}", block_hash))?
+            .response;
+
+        while let Some(event) = special_events.next().await.transpose()? {
+            let has_payday_event = match event {
+                SpecialTransactionOutcome::PaydayPoolReward { .. } => true,
+                SpecialTransactionOutcome::PaydayAccountReward { .. } => true,
+                SpecialTransactionOutcome::PaydayFoundationReward { .. } => true,
+                _ => false,
+            };
+
+            if has_payday_event {
+                return Ok(Some(total_staked_capital));
+            };
+        }
```

Comment:
We should consider adding "is payday block" fucntion to the SDK. Can you add a task there? Or just a PR?

Reply:
I'll add a PR 🙂

Reply:
https://github.com/Concordium/concordium-rust-sdk/pull/66.

Will update this code to use the function when it's merged.